### PR TITLE
[v6r20] DIRAC StorageElement report for free and total disk space

### PR DESCRIPTION
**Pull request by atsareg**
Methods of the DIRAC StorageElement service getFreeDiskSpace() and getTotalDiskSpace() are reimplemented without input arguments. The return values are always in MB. 

The MaxStorageSize CS option is now taken into account when reporting free and total space. The check of available disk space when adding new files is also taking into account the MaxStorageSize, however to avoid massive failures after the update only warning is printed if MaxStorageSize is exceeded while there is still enough available disk space. In the next releases this will become an error 

BEGINRELEASENOTES

*DMS
NEW: StorageElement service - getFreeDiskSpace() and getTotalDiskSpace() take into account MAX_STORAGE_SIZE parameter value

ENDRELEASENOTES

### CODE REVIEW

File: DataManagementSystem/Service/StorageElementHandler.py [hunk 1]
```diff
@@ -74,420 +75,503 @@ def getDiskSpace(path, size = 'TB', total = False):
...
+  result = getTotalDiskSpace()
+  if not result['OK']:
+    return result
+  totalSpace = result['Value']
+  if ignoreMaxStorageSize:
+    maxTotalSpace = totalSpace
+  else:
+    maxTotalSpace = min(totalSpace, MAX_STORAGE_SIZE) if MAX_STORAGE_SIZE else totalSpace
+  occupiedSpace = totalSpace - freeSpace
+  freeSpace = maxTotalSpace - occupiedSpace
```
Comment (fstagni) [inline DataManagementSystem/Service/StorageElementHandler.py]: This is maybe only my taste, but written like this to me is confusing. Can you just calculate and return `freeSpace` inside the `if..else`?

File: DataManagementSystem/Service/StorageElementHandler.py [hunk 2]
```diff
@@ -74,420 +75,503 @@ def getDiskSpace(path, size = 'TB', total = False):
...
+def getFreeDiskSpace(ignoreMaxStorageSize=True):
+  """ Returns the free disk space still available for writing taking into account
+      the total available disk space and the MAX_STORAGE_SIZE limitation
 
-def initializeStorageElementHandler( serviceInfo ):
+      :return: S_OK/S_ERROR, Value is the free space of the SE storage in MB
+  """
+
+  global MAX_STORAGE_SIZE
+  global BASE_PATH
```
Comment (fstagni) [inline DataManagementSystem/Service/StorageElementHandler.py]: Do we really need all these globals? They are considered a not-so-nice practice.

File: DataManagementSystem/Service/StorageElementHandler.py [hunk 3]
```diff
@@ -74,420 +75,503 @@ def getDiskSpace(path, size = 'TB', total = False):
...
+    # We do not have enough space taking into account MAX_STORAGE_SIZE limit
+    # check if the total disk space is enough however
+    result = getFreeDiskSpace(ignoreMaxStorageSize=True)
+    if not result['OK']:
+      return result
+    freeSpace = result['Value']
+    spaceFlag = freeSpace * 1024 * 1024 > size
+    if spaceFlag:
+      gLogger.warn('Space limited by MAX_STORAGE_SIZE is not enough to store file')
+    return S_OK(spaceFlag)
```
Comment (fstagni) [inline DataManagementSystem/Service/StorageElementHandler.py]: Is code here repeated twice?

File: DataManagementSystem/Service/StorageElementHandler.py (hunk 2 above)
Comment (atsareg) [reply DataManagementSystem/Service/StorageElementHandler.py]: I just continue the same style as was in the original file. Re-styling it is not the goal of this PR

**Pull request merged**